src/soc/ibm/power9/mvpd.c

```diff
@@ -65,7 +67,7 @@ struct pt_record {
  * of bytes read. */
 static int read_eeprom_chip(uint8_t cpu, uint32_t offset, void *data, uint16_t len)
 {
-	const unsigned int bus = 1 + 4 * cpu; // four I2C buses per CPU
+	const unsigned int bus = (cpu == 0 ? 1 : FSI_I2C_BUS);
```

Comment:
Making sure: this will always be FSI for second CPU because MVPD is needed before XSCOM is initialized and all later code uses MVPD from cache?

Reply:
Yes, except that it works without caching too (probably slower than it would with XSCOM).